### PR TITLE
Task status validation

### PR DESCRIPTION
**Pull request by danielui**
# Pull Request summary:
Task status API enhancements

# Description of changes:
- Moved TasksAPI to primitives
- Validate task by check it's "Validation" field

# Checklist:
- [X] Make sure to run pre-commit before commiting changes
- [X] Make sure all checks have passed
- [X] PR description is clear and comprehensive
- [X] Mentioned the issue that this PR solves (if applicable)
- [X] Make sure you test the changes

### CODE REVIEW

File: vmngclient/primitives/task_status_api.py
Comment (sbasan) [inline vmngclient/primitives/task_status_api.py:1]: rename file to configuration_dashboard_status.py

File: vmngclient/primitives/task_status_api.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    summary: Summary
+    is_cancel_enabled: bool = Field(alias="isCancelEnabled")
+    is_parallel_execution_enabled: bool = Field(alias="isParallelExecutionEnabled")
+
+
+class TasksData(BaseModel):
+    running_tasks: List[RunningTaskData] = Field(alias="runningTasks")
+
+
+class TasksPrimitives(APIPrimitiveBase):
```
Comment (sbasan) [inline vmngclient/primitives/task_status_api.py:87]: rename class to `ConfigurationDashboardStatusPrimitives` according to:
[ConfigurationDashboardStatusApi](https://github.com/sbasan/vmanage-python-open-api/blob/main/docs/ConfigurationDashboardStatusApi.md)

File: vmngclient/primitives/task_status_api.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    """
+    API class for getting data about tasks
+    """
+
+    def get_task_data(self, task_id) -> TaskData:
```
Comment (sbasan) [inline vmngclient/primitives/task_status_api.py:92]: rename to `find_status` according to:
[ConfigurationDashboardStatusApi](https://github.com/sbasan/vmanage-python-open-api/blob/main/docs/ConfigurationDashboardStatusApi.md)

provide type hint for task_id parameter

**Review by sbasan — CHANGES_REQUESTED** [review]
primitives class needs adjustments

File: vmngclient/api/task_status_api.py [hunk 3]
```diff
@@ -102,6 +28,22 @@ def __init__(self, session: vManageSession, task_id: str):
         self.url = f"/dataservice/device/action/status/{self.task_id}"
         self.task_data: List[SubTaskData]
 
+    def __validate_task(self) -> bool:
+        WAIT_SECONDS = 10
+        REPEATS_NUMBER = 2
+        for _ in range(REPEATS_NUMBER):
+            json = self.session.get_json(self.url)
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:35]: use primitive calls instead raw session requests

File: vmngclient/api/task_status_api.py [hunk 4]
```diff
@@ -102,6 +28,22 @@ def __init__(self, session: vManageSession, task_id: str):
...
+            task_data = TaskData.parse_obj(json)
+            print(task_data)
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:37]: remove print statement

**Review by sbasan — CHANGES_REQUESTED** [review]
additional changess

File: vmngclient/api/task_status_api.py [hunk 5]
```diff
@@ -102,6 +32,21 @@ def __init__(self, session: vManageSession, task_id: str):
         self.url = f"/dataservice/device/action/status/{self.task_id}"
         self.task_data: List[SubTaskData]
 
+    def __validate_task(self):
+        WAIT_SECONDS = 10
+        REPEATS_NUMBER = 2
```
Comment (tehAgitto) [inline vmngclient/api/task_status_api.py:37]: put that as an argument

File: vmngclient/api/task_status_api.py [hunk 6]
```diff
@@ -102,6 +32,21 @@ def __init__(self, session: vManageSession, task_id: str):
...
+        for _ in range(REPEATS_NUMBER):
+            task_data = ConfigurationDashboardStatusPrimitives(self.session).find_status(self.task_id)
+            if task_data.validation.status in (OperationStatus.VALIDATION_SUCCESS, OperationStatus.SUCCESS):
```
Comment (tehAgitto) [inline vmngclient/api/task_status_api.py:40]: can we scale `OperationStatus.VALIDATION_SUCCESS` and  `OperationStatus.SUCCESS` into one `Enum` object?

File: vmngclient/api/task_status_api.py [hunk 7]
```diff
@@ -102,6 +32,21 @@ def __init__(self, session: vManageSession, task_id: str):
...
+                return None
+            elif task_data.validation.status in (OperationStatus.FAILURE, OperationStatus.VALIDATION_FAILURE):
```
Comment (tehAgitto) [inline vmngclient/api/task_status_api.py:42]: can we scale into one `Enum` object?

**Review by tehAgitto — CHANGES_REQUESTED** [review]

File: vmngclient/api/task_status_api.py (hunk 6 above)
Comment (danielui) [reply vmngclient/api/task_status_api.py:40]: Outdated

File: vmngclient/api/task_status_api.py (hunk 5 above)
Comment (danielui) [reply vmngclient/api/task_status_api.py:37]: Done

File: vmngclient/primitives/primitive_container.py [hunk 8]
```diff
@@ -17,3 +18,4 @@ def __init__(self, session: vManageSession):
         self.tenant_management = TenantManagementPrimitives(session)
         self.tenant_backup_restore = TenantBackupRestorePrimitives(session)
         self.tenant_migration = TenantMigrationPrimitives(session)
+        self.configuration_dashboard_status_primitives = ConfigurationDashboardStatusPrimitives(session)
```
Comment (sbasan) [inline vmngclient/primitives/primitive_container.py:21]: variable name without *_primitive suffix to keep it short and consistent with others

File: vmngclient/primitives/task_status_api.py [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        Args:
+            delay_seconds (int, optional): If vmanage doesn't get data about task, after this time will asks again.
+            Defaults to 5.
+
+        Returns:
+            List[SubTaskData]: List of all sub-tusks
+        """
+        url = f"/dataservice/device/action/status/{task_id}"
+        task_data = self.session.get_json(url)
+        return TaskData.parse_obj(task_data)
```
[the remark this answers is not in the document]
Comment (sbasan) [reply vmngclient/primitives/task_status_api.py:105]: resolved? but code and docstrings not changed

[the remark this answers is not in the document]
Comment (danielui) [reply vmngclient/primitives/task_status_api.py]: Changes are in configuration_dashboard_status, this old file. It's already deleted.

File: vmngclient/api/task_status_api.py [hunk 10]
```diff
@@ -145,6 +85,7 @@ def wait_for_completed(
 
             timeout_seconds (int): After this time, function will stop requesting action status
             interval_seconds (int): interval between action status requests
+            wait_seconds (int): After this time, task validation call will be send
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:88]: I think `validation_timeout` is better name for that parameter

File: vmngclient/api/task_status_api.py [hunk 11]
```diff
@@ -102,10 +32,20 @@ def __init__(self, session: vManageSession, task_id: str):
         self.url = f"/dataservice/device/action/status/{self.task_id}"
         self.task_data: List[SubTaskData]
 
+    def __validate_if_failure(self, wait_seconds: int = 5):
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:35]: I think `__check_validation_status` is better name.
Current name suggest that we are validating something when something failed.

**Review by sbasan — CHANGES_REQUESTED** [review]
unclear naming

**Review by sbasan — APPROVED** [review]

File: vmngclient/tests/test_task_status_api.py [hunk 12]
```diff
@@ -84,30 +89,13 @@ def setUp(self):
...
-    @patch("vmngclient.session.vManageSession")
-    def test_wait_for_completed_status_out_of_range(self, mock_session):
-        # Prepare mock data
-        mock_session.get_data.return_value = self.action_data_time_out
-
-        # Assert
-        answer = Task(mock_session, "mock_action_id").wait_for_completed(1, 1).result
-        self.assertEqual(answer, False)
-
+    @unittest.skip("")
```
Comment (cicharka) [inline vmngclient/tests/test_task_status_api.py:92]: Why do we skip this test?

**Review by cicharka — APPROVED** [review]

**Review by tehAgitto — APPROVED** [review]

File: vmngclient/tests/test_task_status_api.py (hunk 12 above)
Comment (tehAgitto) [reply vmngclient/tests/test_task_status_api.py:92]: Lets move that into #268

**Pull request merged**